llvm-general/src/LLVM/General/Internal/Module.hs

```diff
@@ -236,6 +236,19 @@ getDataLayout m = do
   dlString <- decodeM =<< FFI.getDataLayout m
   either fail return . runExcept . parseDataLayout A.BigEndian $ dlString
 
+-- | Build an LLVM.General.'Module' from a
+-- LLVM.General.AST.'LLVM.General.AST.Module' with a 'DataLayout'
+-- appropriate for the given 'TargetMachine'. This is important for
+-- native code generation as it impacts name mangling.
+withModuleFromASTForTargetMachine :: Context -> TargetMachine -> A.Module
```

Comment:
It was use of debatable importance: I simply wanted to lower an `llvm-general-pure` AST to LLVM IR for compilation by other means. The straight line path from `withModuleFromAST` to  `moduleLLVMAssembly` is what tripped me up. You may recall our conversation on this where it turned out to be some externing flag OS X wanted in order to properly compile the generated IR with system tooling.

I understand @bscarlet's position: it is tricky to cover from the `llvm-general` position since it lies in between `llvm-general-pure` and `llvm-general` in a sense. I noticed (and appreciated!) that it is handled by the `OrcJIT` bindings, and I viewed my addition as effectively an IR backend (thinking of OrcJIT as another backend).